compiler/modules/types/string.bal

```diff
@@ -65,6 +65,41 @@ function stringSubtypeContains(SubtypeData d, string s) returns boolean {
     return nonChar.values.indexOf(s) != () ? nonChar.allowed : !nonChar.allowed;
 }
 
+function stringSubtypeContainedIn(SubtypeData d, string[] values) returns boolean {
```

Comment:
This is testing whether the d is a subtype of the set of strings in values, right?

A simpler way to do this is to construct a StringSubtype for the values and then use diff and see if you get false back, which would be O(log(N)) vs what you have here which is O(N^2).

Reply:
Need a function that creates a StringSubtype from a sorted list of strings by splitting into Char and NonChar

Reply:
> This is testing whether the d is a subtype of the set of strings in values, right?
> 
> A simpler way to do this is to construct a StringSubtype for the values and then use diff and see if you get false back, which would be O(log(N)) vs what you have here which is O(N^2).

I mean O(N)